api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortQueryBuilder.java

```diff
@@ -35,25 +80,97 @@
+    queryBuilder.append(DEMO_CHART_INFO_SQL_GROUP_BY.replace("${mainTable}", SEARCH_PERSON_TABLE));
+
+    return QueryJobConfiguration
+      .newBuilder(queryBuilder.toString())
+      .setNamedParameters(params)
+      .setUseLegacySql(false)
+      .build();
+  }
+
+  /**
+   * Provides counts with domain info for charts
+   * defined by the provided {@link ParticipantCriteria}.
+   */
+  public QueryJobConfiguration buildDomainChartInfoCounterQuery(ParticipantCriteria participantCriteria,
+                                                                DomainType domainType,
+                                                                int chartLimit) {
+    String endSqlTemplate = DOMAIN_CHART_INFO_SQL_GROUP_BY
+      .replace("${limit}", Integer.toString(chartLimit))
+      .replace("${tableId}", "standard_concept_id")
+      .replace("${domain}", domainType.name());
+    StringBuilder queryBuilder = new StringBuilder(DOMAIN_CHART_INFO_SQL_TEMPLATE.replace("${mainTable}", QueryBuilderConstants.REVIEW_TABLE));
+    Map<String, QueryParameterValue> params = new HashMap<>();
+    addWhereClause(participantCriteria, QueryBuilderConstants.REVIEW_TABLE, queryBuilder, params);
+    queryBuilder.append(endSqlTemplate.replace("${mainTable}", QueryBuilderConstants.REVIEW_TABLE));
+
+    return QueryJobConfiguration
+      .newBuilder(queryBuilder.toString())
+      .setNamedParameters(params)
+      .setUseLegacySql(false)
+      .build();
+  }
+
+  public QueryJobConfiguration buildRandomParticipantQuery(ParticipantCriteria participantCriteria,
+                                                           long resultSize,
+                                                           long offset) {
+    String endSql = RANDOM_SQL_ORDER_BY + " " + resultSize;
+    if (offset > 0) {
+      endSql += OFFSET_SUFFIX + offset;
+    }
+    String sqlTemplate = RANDOM_SQL_TEMPLATE.replace("${table}", PERSON_TABLE);
+    Map<String, QueryParameterValue> params = new HashMap<>();
+    StringBuilder queryBuilder = new StringBuilder(sqlTemplate.replace("${mainTable}", PERSON_TABLE));
+    addWhereClause(participantCriteria, PERSON_TABLE, queryBuilder, params);
+    queryBuilder.append(endSql.replace("${mainTable}", PERSON_TABLE));
+
+    return QueryJobConfiguration
+      .newBuilder(queryBuilder.toString())
+      .setNamedParameters(params)
+      .setUseLegacySql(false)
+      .build();
   }
 
-  public QueryJobConfiguration buildQuery(ParticipantCriteria participantCriteria,
-      String sqlTemplate, String endSql, String mainTable,
-      Map<String, QueryParameterValue> params) {
-    StringBuilder queryBuilder = new StringBuilder(sqlTemplate.replace("${mainTable}", mainTable));
-    addWhereClause(participantCriteria, mainTable, queryBuilder, params);
-    queryBuilder.append(endSql.replace("${mainTable}", mainTable));
+  //TODO: implemented for use with the Data Set Builder. Please remove it this does not become the preferred solution
```

Comment:
Nit: There's no action TODO here right now, I'd just leave this as a comment without the TODO

Reply:
Done.